cxx20: sequence builder does not work how we use it in align_pairwise

The following code does not compile with `g++-10 -std=c++2a`:

```c++
    auto && sequence1 = this->fst;
    auto && sequence2 = this->sec | std::views::drop(0);

    using first_seq_t = decltype(sequence1) &;
    using second_seq_t = decltype(sequence2) &;

    using function_wrapper_t = std::function<void(first_seq_t, second_seq_t)>;

    auto alignment = [&](first_seq_t sequence1, second_seq_t sequence2) {
      seqan3::detail::aligned_sequence_builder builder{sequence1, sequence2};
      auto trace_path = this->path({seqan3::detail::row_index_type{2},
                                    seqan3::detail::column_index_type{3}});
      auto trace_res = builder(trace_path);
    };

    auto algorithm = function_wrapper_t{std::move(alignment)};
```

With error:
```c++
In file included from /seqan3/test/unit/alignment/matrix/detail/aligned_sequence_builder_test.cpp:13:
/seqan3/include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp: In instantiation of ‘seqan3::detail::aligned_sequence_builder<fst_sequence_t, sec_sequence_t>::result_type seqan3::detail::aligned_sequence_builder<fst_sequence_t, sec_sequence_t>::operator()(trace_path_t&&) [with trace_path_t = std::ranges::subrange<seqan3::detail::trace_iterator<seqan3::detail::two_dimensional_matrix<seqan3::detail::trace_directions>::iterator_type<seqan3::detail::two_dimensional_matrix<seqan3::detail::trace_directions> > >, std::default_sentinel_t, std::ranges::subrange_kind::unsized>&; fst_sequence_t = std::vector<seqan3::dna4>&; sec_sequence_t = std::ranges::drop_view<std::ranges::ref_view<std::vector<seqan3::dna15> > >&]’:
/seqan3/test/unit/alignment/matrix/detail/aligned_sequence_builder_test.cpp:305:31:   required from ‘void aligned_sequence_builder_fixture_issueXXXX_Test<gtest_TypeParam_>::TestBody() [with gtest_TypeParam_ = std::pair<std::vector<seqan3::dna4>&, std::vector<seqan3::dna15>&>]’
/seqan3/test/unit/alignment/matrix/detail/aligned_sequence_builder_test.cpp:291:1:   required from here
/seqan3/include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp:196:25: error: no matching function for call to ‘assign_unaligned(std::__tuple_element_t<1, std::tuple<seqan3::gap_decorator<std::span<seqan3::dna4, 18446744073709551615> >, seqan3::gap_decorator<seqan3::detail::view_take<std::ranges::drop_view<std::ranges::drop_view<std::ranges::ref_view<std::vector<seqan3::dna15, std::allocator<seqan3::dna15> > > > >, false, false> > > >&, std::span<seqan3::dna15, 18446744073709551615>)’
  196 |         assign_unaligned(std::get<1>(res.alignment),
      |         ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
  197 |                          sec_rng | views::slice(res.second_sequence_slice_positions.first,
      |                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  198 |                                                 res.second_sequence_slice_positions.second));
      |                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /seqan3/include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp:18,
                 from /seqan3/test/unit/alignment/matrix/detail/aligned_sequence_builder_test.cpp:13:
/seqan3/include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp:378:13: note: candidate: ‘void seqan3::assign_unaligned(aligned_seq_t&, unaligned_sequence_type&&) [with aligned_seq_t = seqan3::gap_decorator<seqan3::detail::view_take<std::ranges::drop_view<std::ranges::drop_view<std::ranges::ref_view<std::vector<seqan3::dna15> > > >, false, false> >; unaligned_sequence_type = std::span<seqan3::dna15, 18446744073709551615>]’
  378 | inline void assign_unaligned(aligned_seq_t & aligned_seq, unaligned_sequence_type && unaligned_seq)
      |             ^~~~~~~~~~~~~~~~
/seqan3/include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp:378:13: note: constraints not satisfied
In file included from /seqan3/include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp:24,
                 from /seqan3/test/unit/alignment/matrix/detail/aligned_sequence_builder_test.cpp:13:
/seqan3/include/seqan3/range/decorator/gap_decorator.hpp:293:17: note: candidate: ‘void seqan3::assign_unaligned(seqan3::gap_decorator<inner_type>&, unaligned_seq_t&&) [with unaligned_seq_t = std::span<seqan3::dna15, 18446744073709551615>; inner_type = seqan3::detail::view_take<std::ranges::drop_view<std::ranges::drop_view<std::ranges::ref_view<std::vector<seqan3::dna15> > > >, false, false>]’
  293 |     friend void assign_unaligned(gap_decorator & dec, unaligned_seq_t && unaligned)
      |                 ^~~~~~~~~~~~~~~~
/seqan3/include/seqan3/range/decorator/gap_decorator.hpp:293:17: note: constraints not satisfied
```

This is bad, because we normally type erase the alignment algorithm within the configuration:

https://github.com/seqan/seqan3/blob/6d36431dc8611ed0979f0053948672840a7c5fd4/include/seqan3/alignment/pairwise/alignment_configurator.hpp#L340-L341

## Platform

  - SeqAn version: <!-- enter version here -->
  - Operating system: <!-- enter output of `uname -a` here-->
  - Compiler: <!-- enter output of `/path/to/compiler --version` here -->

## Description

### How to repeat the problem

<!-- step to repeat the problem, code snippets, compiler flags... -->

### Expected behaviour

<!-- what you thought would happen -->

### Actual behaviour

<!-- what happens -->

--- a/include/seqan3/alignment/matrix/alignment_trace_algorithms.hpp
+++ b/include/seqan3/alignment/matrix/alignment_trace_algorithms.hpp
@@ -22,6 +22,7 @@
 #include <seqan3/alphabet/gap/gapped.hpp>
 #include <seqan3/core/type_traits/range.hpp>
 #include <seqan3/range/views/slice.hpp>
+#include <seqan3/range/views/type_reduce.hpp>
 
 namespace seqan3::detail
 {
@@ -121,8 +122,10 @@ inline alignment_t alignment_trace(database_t && database,
     assert(coordinate.col < matrix.cols());
 
     alignment_t aligned_seq{};
-    assign_unaligned(std::get<0>(aligned_seq), views::slice(database, front_coordinate.first, coordinate.col));
-    assign_unaligned(std::get<1>(aligned_seq), views::slice(query, front_coordinate.second, coordinate.row));
+    assign_unaligned(std::get<0>(aligned_seq), database | views::type_reduce
+                                                        | views::slice(front_coordinate.first, coordinate.col));
+    assign_unaligned(std::get<1>(aligned_seq), query | views::type_reduce
+                                                     | views::slice(front_coordinate.second, coordinate.row));
     auto end_aligned_db = std::ranges::cend(std::get<0>(aligned_seq));
     auto end_aligned_qy = std::ranges::cend(std::get<1>(aligned_seq));
 

--- a/include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp
+++ b/include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp
@@ -191,11 +191,11 @@ public:
             std::pair<size_t, size_t>{trace_it.coordinate()};
 
         assign_unaligned(std::get<0>(res.alignment),
-                         fst_rng | views::slice(res.first_sequence_slice_positions.first,
-                                                res.first_sequence_slice_positions.second));
+                         std::views::all(fst_rng) | views::slice(res.first_sequence_slice_positions.first,
+                                                                 res.first_sequence_slice_positions.second));
         assign_unaligned(std::get<1>(res.alignment),
-                         sec_rng | views::slice(res.second_sequence_slice_positions.first,
-                                                res.second_sequence_slice_positions.second));
+                         std::views::all(sec_rng) | views::slice(res.second_sequence_slice_positions.first,
+                                                                 res.second_sequence_slice_positions.second));
 
         // Now we need to insert the values.
         fill_aligned_sequence(trace_segments | std::views::reverse,

--- a/test/unit/alignment/matrix/detail/aligned_sequence_builder_test.cpp
+++ b/test/unit/alignment/matrix/detail/aligned_sequence_builder_test.cpp
@@ -7,6 +7,7 @@
 
 #include <gtest/gtest.h>
 
+#include <seqan3/std/algorithm>
 #include <list>
 #include <vector>
 
@@ -18,7 +19,7 @@
 #include <seqan3/alphabet/nucleotide/dna4.hpp>
 #include <seqan3/range/views/to_char.hpp>
 #include <seqan3/range/views/to.hpp>
-#include <seqan3/std/algorithm>
+#include <seqan3/test/expect_range_eq.hpp>
 
 using seqan3::operator|;
 
@@ -286,4 +287,22 @@ TYPED_TEST(aligned_sequence_builder_fixture, second_empty)
     EXPECT_EQ(second_sequence_slice_positions, (std::pair<size_t, size_t>{0u, 0u}));
     EXPECT_EQ(std::get<0>(alignment) | seqan3::views::to_char | seqan3::views::to<std::string>, std::string{"ACG"});
     EXPECT_EQ(std::get<1>(alignment) | seqan3::views::to_char | seqan3::views::to<std::string>, std::string{"---"});
+}
+
+// https://github.com/seqan/seqan3/issues/1921
+TYPED_TEST(aligned_sequence_builder_fixture, issue1921)
+{
+    auto && sequence1 = this->fst;
+    auto && sequence2 = this->sec | std::views::drop(0);
+
+    seqan3::detail::aligned_sequence_builder builder{sequence1, sequence2};
+    auto trace_path = this->path({seqan3::detail::row_index_type{2},
+                                  seqan3::detail::column_index_type{3}});
+
+    auto [first_sequence_slice_positions, second_sequence_slice_positions, alignment] = builder(trace_path);
+
+    EXPECT_EQ(first_sequence_slice_positions, (std::pair<size_t, size_t>{0, 3}));
+    EXPECT_EQ(second_sequence_slice_positions, (std::pair<size_t, size_t>{0, 2}));
+    EXPECT_RANGE_EQ(std::get<0>(alignment) | seqan3::views::to_char, std::string{"--ACG"});
+    EXPECT_RANGE_EQ(std::get<1>(alignment) | seqan3::views::to_char, std::string{"AG---"});
 }

